sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITestBase.scala

```diff
@@ -76,11 +76,15 @@ abstract class SparkE2EQueryITestBase
     item.getAs[String]("id") shouldEqual id
 
     assertMetrics(meterRegistry, "cosmos.client.op.latency", expectedToFind = true)
+
     // Gateway requests are not happening always - but they can happen
     //assertMetrics(meterRegistry, "cosmos.client.req.gw", expectedToFind = true)
+
     assertMetrics(meterRegistry, "cosmos.client.req.rntbd", expectedToFind = true)
     assertMetrics(meterRegistry, "cosmos.client.rntbd", expectedToFind = true)
-    assertMetrics(meterRegistry, "cosmos.client.rntbd.addressResolution", expectedToFind = true)
+
+    // address resolution requests can but don't have to happen - they are optional
+    // assertMetrics(meterRegistry, "cosmos.client.rntbd.addressResolution", expectedToFind = true)
```

Comment:
why this is commented out?

Reply:
Added a comment with explanation - the validation would be flaky